src/browser_launchers/puppeteer_launcher.js

```diff
@@ -64,18 +64,20 @@ export class PuppeteerLauncher extends BrowserLauncher {
     constructor(launchContext = {}) {
         ow(launchContext, 'PuppeteerLauncher', ow.object.exactShape(PuppeteerLauncher.optionsShape));
 
-        launchContext.launcher = launchContext.launcher || require('puppeteer'); // eslint-disable-line
         const {
+            launcher = BrowserLauncher.requireLauncherOrThrow('puppeteer'),
             userAgent,
             stealth,
             stealthOptions,
             ...browserLauncherOptions
         } = launchContext;
 
+        browserLauncherOptions.launcher = launcher;
+
         super(browserLauncherOptions);
 
         this.userAgent = userAgent;
-        this.stealth = stealth; // @TODO: time to rename to useStealth?
+        this.stealth = stealth;
```

Comment:
Here it's fine I guess, because the `browserLauncherOptions` is a newly created object. But I'd still do the `super({ ...browserLauncherOptions, launcher })` for consistent code style.